patches/conf-libclang/configure-generic.sh.1

```diff
@@ -0,0 +1,142 @@
+                ;;
+            brew)
+                llvm_config="$(brew --prefix)/opt/llvm/bin/llvm-config" || continue
+                ;;
+            none)
+                break
+                ;;
+            esac
+            llvm_version="$($llvm_config --version)" || continue
+            next_version=$((maximum_version + 1))
+            if [\
+                $(printf "${llvm_version%%.*}\n$next_version" | sort -n | head -n1)\
+                    = $next_version ]; then
+                continue
+            fi
+            break
+        done
+	if [ "$kind" == none ]; then
+	    continue
+	fi
+    else
+        if hash brew 2>/dev/null; then
+           brew_llvm_config="$(brew --cellar llvm)"/${version}*/bin/llvm-config || true
+           brew_llvm_config_at="$(brew --cellar llvm@${version})"/${version}*/bin/llvm-config || true
+        fi
+        for llvm_config in \
+            llvm-config-${version} llvm-config-${version}.0 \
+            llvm-config${version}0 llvm-config${version} \
+            llvm-config-${version}-32 llvm-config-${version}-64 \
+            llvm-config-mp-$version \
+            llvm-config-mp-${version}.0 $brew_llvm_config $brew_llvm_config_at \
+            /usr/lib64/llvm/${version}/bin/llvm-config \
+            /usr/lib/llvm/${version}/bin/llvm-config \
+            /usr/lib/llvm${version}/bin/llvm-config; do
+            llvm_config="$(command -v $llvm_config)" || continue
+            llvm_version="$($llvm_config --version)" || continue
+            break
+        done
+        if [ -z "$llvm_version" ]; then
+            continue
+        fi
+    fi
+
+    LLVM_CFLAGS="$($llvm_config --cflags)"
+    LLVM_LDFLAGS="$($llvm_config --ldflags)"
+    LLVM_LIBDIR="$($llvm_config --libdir)"
+
+    # These filters enable compilation with gcc.
+    # Filter -Wstring-conversion for OpenSUSE
+    LLVM_CFLAGS="$(echo $LLVM_CFLAGS | sed 's/-Wstring-conversion //')"
+
+    # Filter -Werror=unguarded-availability-new and -Wcovered-switch-default
+    # (which appear with LLVM 7)
+    LLVM_CFLAGS="$(echo $LLVM_CFLAGS | sed 's/-Werror=unguarded-availability-new //')"
+    LLVM_CFLAGS="$(echo $LLVM_CFLAGS | sed 's/-Wcovered-switch-default //')"
+
+    # Filter "-Wdelete-non-virtual-dtor" (warning only)
+    LLVM_CFLAGS="$(echo $LLVM_CFLAGS | sed 's/-Wdelete-non-virtual-dtor //')"
+
+    tempdir="$(mktemp -d)"
+    cat >"$tempdir/test_libclang.c" <<EOF
+#include <clang-c/Index.h>
+#include <stdlib.h>
+
+int
+main(int argc, char *argv[])
+{
+  CXIndex idx = clang_createIndex(1, 1);
+  clang_disposeIndex(idx);
+  return EXIT_SUCCESS;
+}
+EOF
+
+    CC=cc
+    if "$CC" -o "$tempdir/test_libclang.o" -c $LLVM_CFLAGS \
+        "$tempdir/test_libclang.c" &&
+      "$CC" -o "$tempdir/test_libclang" \
+          $LLVM_LDFLAGS "$tempdir/test_libclang.o" \
+          "-lclang" "-Wl,-rpath,$LLVM_LIBDIR" &&
+      "$tempdir/test_libclang"; then
+        true
+    else
+        clean_tempdir
```

Comment:
instead of manually ensuring that clean_tmpdir is called, you can use "trap clean_tmpdir EXIT" at top level after clean_tmpdir is defined.